low-level/message_handlers/node_data_msg_handler.py

```diff
@@ -671,6 +757,9 @@ def _get_nwalert(self, interfaces):
             for interface in interfaces:
                 interface_name = interface.get("ifId")
                 nw_status = interface.get("nwStatus")
+                if interface_name == pvt_data_nw_if_name and self.pvt_data_nw_conn_unknown_fault:

```

Comment:
why we are not handling generic fault generation here itself, what is being done from line 729 - 744 ?